privacyidea/models.py

```diff
@@ -1512,14 +1512,14 @@ def __str__(self):
         return "{0!s}".format(descr)
 
 
-def cleanup_challenges():
+def cleanup_challenges(serial):
     """
     Delete all challenges, that have expired.
 
     :return: None
     """
     c_now = datetime.utcnow()
-    Challenge.query.filter(Challenge.expiration < c_now).delete()
+    Challenge.query.filter(Challenge.expiration < c_now, Challenge.serial == serial).delete()
```

Comment:
I am not sure if that solves the problem since we still have the `expiration < c_now` in the query. But i'll merge this for now.